### PR TITLE
Additional 1.2 changes

### PR DESCRIPTION
**Pull request by blowfishpro**
RF compiles and runs on KSP 1.2.2 with this.  Might be bugs I haven't found yet.

### CODE REVIEW

**Comment by Starwaster** [comment]
How much more do you need to do before you merge the PR?

**Comment by blowfishpro** [comment]
This is all the changes I am aware need to happen.  Just wanted to let some other people take a look before merging, particularly since I'm relatively new to working with RF

File: Source/Tanks/ModuleFuelTanks.cs [hunk 1]
```diff
@@ -18,6 +18,8 @@ public class ModuleFuelTanks : PartModule, IModuleInfo, IPartCostModifier, IPart
 	{
 		bool compatible = true;
 
+        public readonly EventVoid onUpdateTankType = new EventVoid("onUpdateTankType");
```
Comment (blowfishpro) [inline Source/Tanks/ModuleFuelTanks.cs]: One other option here would be to use a part event.  That would less prone to NREs, and would mean that that the TF module doesn't need to know about the tank directly, but also harder to detect if it breaks.

**Comment by NathanKell** [comment]
One thought I had: what about making ModuleFuelTank a partial class? And put only the base MFT stuff in one file, and RF's bits in another? That would avoid monobehavior spam (bad for performance) while still allowing @taniwha's makefile to skip the stuff he finds icky.

**Comment by blowfishpro** [comment]
Is adding additional MonoBehaviours to the scene really that bad for performance?

**Comment by NathanKell** [comment]
Not the worst thing ever, but it means more stuff for Unity to poll on every update pass IIRC, and more things to look for whenever trying to find a module on a part/vessel (that's O(n) ), which gets done a lot in KSP/mods.

**Comment by blowfishpro** [comment]
I don't have a strong sense of what the performance impact would be.  You've worked with Unity a lot more than I have, so if you think it would be better to implement as partial classes, I will do that.

**Comment by NathanKell** [comment]
The performance hit shouldn't be too bad at all; the question is what's easier to deal with code-wise IMO. If it's a pain to juggle 3 components, partial class is a way out (a way back to how it was).

**Comment by blowfishpro** [comment]
I decided you're right - I would have liked to keep separate classes, but there's just too much state that they share so it doesn't really make sense in this case.

Have noticed that ullage state seems to be stickier than I remember it - i.e. if it's stable then it will stay stable even with no acceleration.  Haven't been able to trace it to any of my changes though.  Any thoughts on this?

**Comment by Starwaster** [comment]
Do you notice that behavior on all tanks or just cryogenic ones? Boiloff mass vented from cryotanks is used for ullage control.

**Comment by blowfishpro** [comment]
Ah, didn't realize that had been implemented.  I was testing with cryogenics.

**Comment by Starwaster** [comment]
That's actually been in for awhile. Something else that's in is that if tankage is allocated for the boiloff gas then that is generated and some or all of that will be subtracted from boiloff mass. So far only Service Modules have that feature. (they're the ones most likely to need it)

**Pull request merged**